subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -374,8 +374,9 @@ static int do_update_reply_cb(const struct coap_packet *response,
 	LOG_ERR("Failed with code %u.%u. Retrying registration",
 		COAP_RESPONSE_CODE_CLASS(code),
 		COAP_RESPONSE_CODE_DETAIL(code));
-	set_sm_state(ENGINE_DO_REGISTRATION);
 
+	set_sm_state(ENGINE_DO_REGISTRATION);
```

Comment:
Intuitively it should be first callback, then set state, no?

Reply:
If you look at how they do it in this function they do change the state then finally do some event call. As I did it would be the same thing.

Might be that there should be another state? dunno

Reply:
nah if it follows the existing pattern then good